backend/src/main/java/org/example/restaurantmanagementapplication/rest/OrderController.java

```diff
@@ -41,4 +41,24 @@ public ResponseEntity<Order> createOrder(@RequestBody List<OrderItemRequest> ite
     Order order = orderService.createOrder(orderRequest);
     return ResponseEntity.ok(order);
   }
-}
\ No newline at end of file
+
+  @PostMapping("{id}/cancel")
+  public ResponseEntity<Order> cancelOrder(@PathVariable int id) {
+    Optional<User> currentUser = sessionManager.getCurrentUser();
+
+    var updatedBy = currentUser.isPresent()
+        ? currentUser.get().getEmail()
+        : "anonymous";
+
+    Order order = orderService.findById(id);
+    if (order == null) {
+      return ResponseEntity.notFound().build();
+    } else if (!OrderStatusTransition.isCancellable(order.getStatus())) {
```

Comment:
-- Minor, can be ignored for now and fixed later as it's just cosmetic

In case of `true` we go to the exitpoint, in this case we are good to remove `else`

```suggestion
    } if (!OrderStatusTransition.isCancellable(order.getStatus())) {
```

Reply:
Thanks! I made the suggested change but split the if onto a new line to keep clarity: https://github.com/barankonrad/WPZ-ResturantManagementApp/pull/25/commits/9dedc39f29e8abc56ca6c428dc0f02876b02f229